Change log for 2.0.0 (#10177)
open-telemetry/opentelemetry-java-instrumentation: CHANGELOG.md

## Unreleased

> The 2.0 release contains significant breaking changes that will most likely affect all users,

Reviewer: nit:
```suggestion
The 2.0.0 release contains significant breaking changes that will most likely affect all users,
```

please be sure to read the migration notes below carefully.

Note: 1.32.x will be security patched for at least 6 months in case some of the changes below are
too disruptive to adopt right away.

> ### Migration notes

Reviewer 2: Should we perhaps name the header "Breaking changes"? With some extra ⚠️  emojis as well, so that it catches the eye

- Stable HTTP semantic conventions are now emitted.
  - MANY CHANGES, see https://opentelemetry.io/blog/2023/http-conventions-declared-stable/#summary-of-changes

> - Stable JVM semantic conventions are now emitted.

Reviewer 3: You think its worthwhile to make a note about attribute keys changing as well? Maybe we don't elaborate the the full set of changes, but mention that all have changed and to go read the spec for the new values..

Author: makes sense, I added metric attribute changes as well

  - `process.runtime.jvm.system.cpu.utilization` renamed to `jvm.system.cpu.utilization` (still experimental)
  - `process.runtime.jvm.cpu.utilization` renamed to `jvm.cpu.recent_utilization`
  - `process.runtime.jvm.gc.duration` renamed to `jvm.gc.duration`
  - `process.runtime.jvm.memory.init` renamed to `jvm.memory.init` (still experimental)
  - `process.runtime.jvm.memory.usage` renamed to `jvm.memory.used`
  - `process.runtime.jvm.memory.committed` renamed to `jvm.memory.committed`
  - `process.runtime.jvm.memory.limit` renamed to `jvm.memory.limit`
  - `process.runtime.jvm.memory.usage_after_last_gc` renamed to `jvm.memory.used_after_last_gc`
  - `process.runtime.jvm.threads.count` renamed to `jvm.threads.count`
  - `process.runtime.jvm.buffer.limit` renamed to `jvm.buffer.memory.limit` (still experimental)
  - `process.runtime.jvm.buffer.count` renamed to `jvm.buffer.count` (still experimental)
  - `process.runtime.jvm.buffer.usage` renamed to `jvm.buffer.memory.usage` (still experimental)
- The default OTLP protocol has been changed from `grpc` to `http/protobuf` in order to align with

>   the specification. You can switch to the `grpc` protocol using `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`

Reviewer: ```suggestion
  the 
  [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/0127b1f8f61656cc97db5a7914e6f70d4bde8273/specification/protocol/exporter.md#specify-protocol). 
  You can switch to the `grpc` protocol using `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`
```

## Unreleased

The 2.0.0 release contains significant breaking changes that will most likely affect all users,

> please be sure to read the migration notes below carefully.

Author: ```suggestion
please be sure to read the breaking changes below carefully.
```

  the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.28.0/specification/protocol/exporter.md#specify-protocol).
  You can switch to the `grpc` protocol using `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`
  or `-Dotel.exporter.otlp.protocol=grpc`.
- Micrometer metric bridge has been disabled by default. You can enable it using
  `OTEL_INSTRUMENTATION_MICROMETER_ENABLED=true`
  or `-Dotel.instrumentation.micrometer.enabled=true`.
- The OTLP logs exporter is now enabled by default. You can disable it using
  `OTEL_EXPORTER_LOGS=none` or `-Dotel.exporter.logs=none`.
- Controller spans are now disabled by default. You can enable them using
  `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_CONTROLLER_TELEMETRY_ENABLED=true`
  or `-Dotel.instrumentation.common.experimental.controller-telemetry.enabled=true`.
- View spans are now disabled by default. You can enable them using
  `OTEL_INSTRUMENTATION_COMMON_EXPERIMENTAL_VIEW_TELEMETRY_ENABLED=true`
  or `-Dotel.instrumentation.common.experimental.view-telemetry.enabled=true`.
- ⚠️⚠️ Stable HTTP semantic conventions are now emitted ⚠️⚠️
  - TOO MANY CHANGES TO LIST HERE, be sure to review the full

>     [list of changes](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/migration-guide.md#summary-of-changes).

Reviewer 4: ❤️

## Unreleased

The 2.0.0 release contains significant breaking changes that will most likely affect all users,
please be sure to read the breaking changes below carefully.

Note: 1.32.x will be security patched for at least 6 months in case some of the changes below are
too disruptive to adopt right away.

### ⚠️⚠️ Breaking changes ⚠️⚠️

- The default OTLP protocol has been changed from `grpc` to `http/protobuf` in order to align with
  the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.28.0/specification/protocol/exporter.md#specify-protocol).
  You can switch to the `grpc` protocol using `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`
  or `-Dotel.exporter.otlp.protocol=grpc`.
- Micrometer metric bridge has been disabled by default. You can enable it using
  `OTEL_INSTRUMENTATION_MICROMETER_ENABLED=true`
  or `-Dotel.instrumentation.micrometer.enabled=true`.
- The OTLP logs exporter is now enabled by default. You can disable it using

>   `OTEL_EXPORTER_LOGS=none` or `-Dotel.exporter.logs=none`.

Reviewer 5: ```suggestion
  `OTEL_LOGS_EXPORTER=none` or `-Dotel.logs.exporter=none`.
```
